src/classes/BDI_DataImportService.cls

```diff
@@ -607,6 +629,144 @@ global with sharing class BDI_DataImportService {
+            accessibleDataImports.addAll(dataImportsFLSValidatedAccessible);
+            return accessibleDataImports;
+        }
+
+        //Used for normal mode
+        public List<DataImport__c> getDataImportsWithUpdateableAndCreateableTargetFields() {
+            return dataImportsFLSValidatedCreateUpdate;
+        }
+
+        public List<DataImport__c> getDataImportsWithFlsExceptions(Boolean isDryRunEnabled) {
+            if (isDryRunEnabled) {
+                return dataImportsWithFLSExceptions;
+            } else {
+                List<DataImport__c> dataImportsWithFlsCreateUpdateExceptions = new List<DataImport__c>();
+                dataImportsWithFlsCreateUpdateExceptions.addAll(dataImportsWithFLSExceptions);
+                dataImportsWithFlsCreateUpdateExceptions.addAll(dataImportsFLSValidatedAccessible);
+                return dataImportsWithFlsCreateUpdateExceptions;
+            }
+        }
+
+        //Used to stamp errors
+        public Set<String> getFieldsWithoutCreateUpdateAccess(Id dataImportId) {
+            Set<String> fieldsWithoutCreateUpdateAccess = new Set<String>();
+            fieldsWithoutCreateUpdateAccess.addAll(nonCreateableTargetFieldsByDataImportId.get(dataImportId));
+            fieldsWithoutCreateUpdateAccess.addAll(nonUpdateableTargetFieldsByDataImportId.get(dataImportId));
+            return fieldsWithoutCreateUpdateAccess;
+        }
+
+        //Used to stamp errors in dry run
+        public Set<String> getFieldsWithoutReadAccess(Id dataImportId) {
+            return nonAccessibleTargetFieldsByDataImportId.get(dataImportId);
+        }
+
+        void validateTargetMappingFieldLevelSecurity() {
+            for (DataImport__c dataImport : dataImportsInitial) {
+                Set<String> nonCreateableTargetFields = new Set<String>();
+                Set<String> nonUpdateableTargetFields = new Set<String>();
+                Set<String> nonAccessibleTargetFields = new Set<String>();
+
+                for (String dataImportFieldName : dataImport.getPopulatedFieldsAsMap().keySet()) {
+                    Map<String, Set<String>> targetFieldsByObject = sourceFieldToObjectToFieldsMap.get(dataImportFieldName);
+
+                    if (targetFieldsByObject == null) { //this source field is unmapped
+                        continue;
+                    }
+
+                    //Now loop through all of the target object + field combos for this source field
+                    for (String targetObjectName : targetFieldsByObject.keySet()) {
+                        for (String targetFieldName : targetFieldsByObject.get(targetObjectName)) {
+                            DescribeFieldResult fieldDescribe = UTIL_Describe.getFieldDescribe(targetObjectName, targetFieldName);
+                            if (fieldDescribe.isAccessible() == false) {
+                                nonAccessibleTargetFields.add(targetObjectName + '.' + targetFieldName);
```

Comment:
If we define the string `targetObjectName + '.' + targetFieldName` before we hit these conditionals, I think we can reduce our memory footprint, and possible save a little bit of execution time by rebuilding the same field twice. Technically we might even be able to define our combined field strings outside of the DI loop (or else check against existence in a Set and only create new if its not present already)? As long as it's not an issue sharing instances of that string between collections, but I don't think we are modifying the strings after this point. And string arithmetic is rumored to be relatively expensive?